### PR TITLE
API Generator: Generate field resolver for root blocks

### PR DESCRIPTION
**Pull request by nsams**
This allows skipping the `@Field` annotation for root blocks in the entity and it doesn't need the field middleware anymore.

### CODE REVIEW

File: packages/api/cms-api/src/generator/generate-crud.ts [hunk 1]
```diff
@@ -891,11 +920,18 @@ function generateResolver({ generatorOptions, metadata }: { generatorOptions: Cr
         }
     }
 
+    imports.push({ name: "extractGraphqlFields", importPath: "@comet/cms-api" });
+    imports.push({ name: "SortDirection", importPath: "@comet/cms-api" });
+    imports.push({ name: "RequiredPermission", importPath: "@comet/cms-api" });
+    imports.push({ name: "AffectedEntity", importPath: "@comet/cms-api" });
+    imports.push({ name: "validateNotModified", importPath: "@comet/cms-api" });
+    imports.push({ name: "RootBlockDataScalar", importPath: "@comet/cms-api" });
+    imports.push({ name: "BlocksTransformerService", importPath: "@comet/cms-api" });
```
Comment (johnnyomair) [inline packages/api/cms-api/src/generator/generate-crud.ts:929]: Why this change? And why not do it for all imports? 😁

File: demo/api/src/news/generated/news.resolver.ts [hunk 2]
```diff
@@ -21,6 +29,8 @@ export class NewsResolver {
         private readonly entityManager: EntityManager,
         private readonly newsService: NewsService,
         @InjectRepository(News) private readonly repository: EntityRepository<News>,
+
```
Comment (johnnyomair) [inline demo/api/src/news/generated/news.resolver.ts:32]: Nit: Could we remove this empty line?

File: .changeset/pretty-kangaroos-repair.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+"@comet/cms-api": minor
+---
+
+API Generator: generate field resolver for root blocks
+
+This allows skipping the `@Field` annotation in entity for root blocks
+and it doesn't need the field middleware anymore
```
Comment (johnnyomair) [inline .changeset/pretty-kangaroos-repair.md:8]:
```suggestion
API Generator: Generate field resolver for root blocks

This allows skipping the `@Field` annotation for root blocks in the entity and it doesn't need the field middleware anymore.
```

File: packages/api/cms-api/src/generator/generate-crud.ts (hunk 1 above)
Comment (nsams) [reply packages/api/cms-api/src/generator/generate-crud.ts:929]: For some reason eslint autofix failed to merge duplicated imports (from `@comet/cms-api`) for those, for the others it worked correctly.

> And why not do it for all imports?

yes, maybe we should switch to that technique

**Review by johnnyomair — APPROVED** [review]

**Comment by johnnyomair** [comment]
@nsams please fix conflicts

**Review by johnnyomair — DISMISSED** [review]

**Comment by johnnyomair** [comment]
@nsams the lint workflow fails due to changes in the generated files (I don't understand why they happen though)

**Review by johnnyomair — APPROVED** [review]
